backend/src/main/java/capstone/myfinancemanager/manager/controller/TransactionController.java

```diff
@@ -26,9 +31,15 @@ public List<TransactionDto> getAllTransactions() {
     @ResponseStatus(code = HttpStatus.CREATED)
     @PostMapping
     public ResponseEntity<TransactionDto> addTransaction(@RequestBody TransactionCreationDto transactionCreation) {
+
+        Principal principal = SecurityContextHolder.getContext().getAuthentication();
+        if (principal.getName() == null) {
```

Comment:
When we have no logged in user the principal should be null? Please verify. A better way would be to require authentication in the security config for `POST /transaction`.
Custom exception with speaking name is a good idea!